### ISSUE TITLE
Availability of ifx?

### ISSUE THREAD

**Issue by h-vetinari** [conda-forge/intel-compiler-repack-feedstock#15]
Hey all

Since this feedstock repacks `intel-fortran-rt`, or:
> Runtime for Intel® Fortran Compiler Classic and Intel® Fortran Compiler (Beta)

I'm wondering if it's also planned to repack `ifort` for conda-forge? For scipy, we've been stuck with using an unholy combination of MSVC+gfortran because there's no other available fortran compiler on windows in conda-forge. Now that scipy is switching to meson (which enforces a cleaner toolchain than this hack; [details](https://github.com/conda-forge/scipy-feedstock/issues/213)), we're really in need of a fortran compiler, and llvm-flang resp. LFortran aren't ready for primetime yet.

The reason I'm asking is that I cannot really imagine the usefulness of the intel fortran runtime in conda-forge, unless there are packages that require it, which would presumably be those that have been compiled with `ifort`.

Thanks!

**Comment by napetrov** [comment]
Hi @h-vetinari,
thanks for inquiry. 
Runtime 
1. Think this is just part of all compiler runtimes distribution including C++ and SYCL.
2. Initial aim was to provide runtimes for other applications build based on Intel Compilers so runtime dependency can be resolved via this package.  So non necessary conda-forge build ecosystem but user ecosystem that relay on this runtime packages.
 
Compiler:
1. Compiler itself is publicly available via apt/yum channels but there are additional steps were required for legal part of redistribution to conda. This was done for C++ and SYCL compilers. And technically can be done.
2. ifort is old compiler that wouldn't be developed further, so it for sure wouldn't be deployed.  But ifx is current replacement for ifort - so it probably have chances to get here if there would be enough interest for it. 

This wouldn't be fast...

**Comment by h-vetinari** [comment]
Hi @napetrov, thanks for the reply!

> But ifx is current replacement for ifort - so it probably have chances to get here if there would be enough interest for it.

I wasn't aware how the new Fortran compiler is called,  but this would be hugely helpful! Please let me know what I can do to make that case (including gathering feedback or support from conda-forge/core)

For context: We've had big problems serving packages with Fortran components (mainly on windows), because we have no compilers available for that in conda-forge, except gfortran-through-MinGW, which comes with a long list of headaches though.

Having a production grade compiler available in our infrastructure would be incredibly beneficial.

**Comment by rgommers** [comment]
> Having a production grade compiler available in our infrastructure would be incredibly beneficial.

+1. I'd also add that this would make it a lot easier to use ifx in CI of mainstream packages like SciPy, especially on Windows. The separate installers and `.bat` scripts that Intel provides aren't really workable; a simple `conda install ifx` (or `intel-compilers`) would be way easier. As a data point, I did put time into that at https://github.com/scipy/scipy/pull/16957, but it was too painful to finish.

**Comment by h-vetinari** [comment]
Since the compiler sources are not available, only intel can provide `ifx`; we have no way to build it ourselves. In the meantime, you can try using `flang` 17 or 18 as a MSVC-compatible compiler on windows. It's what we're using for SciPy for example ([background](https://labs.quansight.org/blog/building-scipy-with-flang)).

**Comment by Krande** [comment]
Thank you for the response! 

Okay, then I'll shift my efforts to the flang compiler. 

(A really interesting article btw.)

Best Regards
Kristoffer

**Comment by jacobwilliams** [comment]
Note: it appears that Intel now has the `ifx` compiler available on their conda channel. It's `ifx_linux-64` in `https://software.repos.intel.com/python/conda/`

see: https://www.intel.com/content/www/us/en/developer/tools/oneapi/fortran-compiler-download.html?operatingsystem=linux&distribution-linux=conda

**Comment by jcapriot** [comment]
Windows as well as `ifx_win-64` !

> Note: it appears that Intel now has the `ifx` compiler available on their conda channel. It's `ifx_linux-64` in `https://software.repos.intel.com/python/conda/`
> 
> see: https://www.intel.com/content/www/us/en/developer/tools/oneapi/fortran-compiler-download.html?operatingsystem=linux&distribution-linux=conda

**Comment by Krande** [comment]
But I am guessing that the availability of ifx on intel's conda channel doesnt get things any closer for us to use it for conda-forge packages?

**Comment by napetrov** [comment]
> But I am guessing that the availability of ifx on intel's conda channel doesnt get things any closer for us to use it for conda-forge packages?

yes. actually IFX was available on Intel channel for a while, so there is nothing changed dramatically on that front.  Would try running another interaction internally to see if things get changes from internal perspective

**Comment by napetrov** [comment]
We got green light for deploying IFX to conda-forge.  Will solve legal aspect with NUMFOCUS as was done with dpcpp compiler and then IFX would be added to feedstock.

**Comment by Krande** [comment]
@napetrov that's fantastic news!

**Comment by ekomarova** [comment]
Windows part is ready and will be available on the channel in the very near future, for linux part we will create a separate PR

**Comment by napetrov** [comment]
Ok, Linux part is also done and bits would be available after CI completion.
